pkgs/applications/misc/remnote/default.nix

```diff
@@ -1,14 +1,45 @@
-{ lib, fetchurl, appimageTools }:
+{ lib, fetchurl, appimageTools, stdenv, makeDesktopItem }:
 
-appimageTools.wrapType2 rec {
+let
   pname = "remnote";
-  version = "1.12.9";
-
+  version = "1.12.18";
   src = fetchurl {
     url = "https://download.remnote.io/remnote-desktop/RemNote-${version}.AppImage";
-    sha256 = "sha256-ZBo7yxbTS+2pWecbPGxp0UMy16HRMwuuUUejb6DUHic=";
+    sha256 = "sha256-z/LTsX65rN+AJbpCTzIabpgUadFe6SnRRDKOkexZyjQ=";
   };
-
+  icon = fetchurl {
+    url = "https://www.remnote.io/icon.png";
+    sha256 = "sha256-r5D7fNefKPdjtmV7f/88Gn3tqeEG8LGuD4nHI/sCk94=";
+  };
+  appexec = appimageTools.wrapType2 {
+    inherit pname version src;
+  };
+  desktopItem = makeDesktopItem {
+    type = "Application";
+    name = "remnote";
+    desktopName = "RemNote";
+    comment = "Spaced Repetition";
+    icon = "remnote";
+    exec = "remnote %u";
+    categories = [ "Office" ];
+    mimeTypes = [ "x-scheme-handler/remnote" "x-scheme-handler/rn" ];
+  };
+in
+stdenv.mkDerivation {
+  inherit pname version src;
+  dontUnpack = true;
+  dontConfigure = true;
+  dontBuild = true;
+  installPhase = ''
+    runHook preInstall
+    mkdir -p $out/bin
+    install ${appexec}/bin/remnote-${version} $out/bin/remnote
+    mkdir -p $out/share/applications
+    install -D "${desktopItem}/share/applications/"* -t $out/share/applications/
```

Comment:
I believe that `desktopItem` is a valid attribute for `mkDerivation`, which would make that unnecessary. Could you check?

Reply:
> I believe that `desktopItem` is a valid attribute for `mkDerivation`, which would make that unnecessary. Could you check?

Hmm. I couldn't find any documentation to that effect, and my googling was fruitless. I tried adding it within `mkDerivation`, but couldn't get a desktop file to output. I basically copied the method used in the obsidian build file:

https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/misc/obsidian/default.nix

Reply:
Okay!